Fix #10306 Added "async" keyword to asynchronous functions that return a promise to question-backend-api.service.ts

abhishekalbert opened the pull request:
## Overview
<!--
READ ME FIRST:
Please answer *both* questions below and check off every point from the Essential Checklist!
If there is no corresponding issue number, fill in N/A where it says [fill_in_number_here] below in 1.
-->

1. This PR fixes or fixes part of #[#10306].
2. This PR does the following: Added "async" keyword to asynchronous functions that return a promise to question-backend-api.service.ts and other files from where function can get the call

## Essential Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The linter/Karma presubmit checks have passed locally on your machine.
- [x] "Allow edits from maintainers" is checked. (See [here](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork) for instructions on how to enable it.)
  - This lets reviewers restart your CircleCI tests for you.
- [x] The PR is made from a branch that's **not** called "develop".

## Proof that changes are correct

<!--
Add videos/screenshots of the user-facing interface to demonstrate that the changes made in this PR work correctly.
-->

## PR Pointers

- Make sure to follow the [instructions for making a code change](https://github.com/oppia/oppia/wiki/Contributing-code-to-Oppia#instructions-for-making-a-code-change).
- Oppiabot will notify you when you don't add a PR_CHANGELOG label. If you are unable to do so, please @-mention a code owner (who will be in the Reviewers list), or ask on [Gitter](https://gitter.im/oppia/oppia-chat).
- For what code owners will expect, see the [Code Owner's wiki page](https://github.com/oppia/oppia/wiki/Oppia%27s-code-owners-and-checks-to-be-carried-out-by-developers).
- Make sure your PR follows conventions in the [style guide](https://github.com/oppia/oppia/wiki/Coding-style-guide), otherwise this will lead to review delays.
- Never force push. If you do, your PR will be closed.
- Oppiabot can assign anyone for review/help if you leave a comment like the following: "{{Question/comment}} @{{reviewer_username}} PTAL"
- Some of the e2e tests are flaky, and can fail for reasons unrelated to your PR. We are working on fixing this, but in the meantime, if you need to restart the tests, please check the ["If your build fails" wiki page](https://github.com/oppia/oppia/wiki/If-your-build-fails).

abhishekalbert committed:
Added "async" keyword to asynchronous function that return a promise to file  question-backend-api.services.ts

abhishekalbert committed:
Edited the calling functions in practice-tab.components.ts and question-list.services.ts

Renamed the fetchTotalQuestionCountForSkillIds to fetchTotalQuestionCountForSkillIdsAsync in practice-tab.component.ts and fetchQuestionSummaries tofetchQuestionSummariesAsync in question-list.service.ts

abhishekalbert committed:
Renamed the function

Renamed the function  fetchTotalQuestionCountForSkillIds to fetchTotalQuestionCountForSkillIdsAsync

abhishekalbert committed:
end to end test

protractor screenshots of end to end test

abhishekalbert committed:
Merge pull request #1 from oppia/develop

new pull

nithusha21 commented on core/templates/pages/topic-viewer-page/practice-tab/practice-tab.component.spec.ts:
```diff
@@ -58,7 +58,7 @@ class MockWindowRef {
 
 class MockQuestionBackendApiService {
-  fetchTotalQuestionCountForSkillIds() {
+  fetchTotalQuestionCountForSkillIdsAsync() {
```
```suggestion
  async fetchTotalQuestionCountForSkillIdsAsync() {
```

nithusha21 reviewed (commented):
Just a nit for my codeowner files

kevintab95 reviewed (changes requested):
@abhishekalbert there seem to be a lot of unrelated files added in this PR. Can you please drop those and assign me for review?

abhishekalbert commented:
@kevintab95: I think you are talking about the images files, sir?? 
I added those images because it generated them in the end to end test and first I commit them without these files it showed fail end to end test because of there is no uploading files of protractor screenshots of end to end test that's why I run the end to end test and added those files in Oppia directory named as protractor screenshots. 


sajalasati commented:
@abhishekalbert Please merge from develop, and also don't forget to assign the right people before waiting on them. 
Assigning @kevintab95 as per your earlier comment.

kevintab95 commented:
Thanks @sajalasati for assigning me.

@abhishekalbert there seem to be quite a few files that are not expected in this PR (see report.html and some PNG files). Please only keep files that are related to your changes (i.e. adding `async` / `..Async`) in this PR. Also, once this is done, feel free to assign me for a review. Thanks!

abhishekalbert committed:
sync forked repo with original repo

abhishekalbert committed:
Delete protractor-screenshots directory

abhishekalbert committed:
Merge branch 'develop' into AddAsyncKeyword

abhishekalbert commented:
> @abhishekalbert there seem to be a lot of unrelated files added in this PR. Can you please drop those and assign me for review?

Fixed @kevintab95

abhishekalbert committed:
Merge pull request #2 from abhishekalbert/AddAsyncKeyword

Add async keyword

abhishekalbert committed:
Merge pull request #3 from oppia/develop

main oppia pull

abhishekalbert committed:
Merge pull request #4 from abhishekalbert/develop

new changes pull

abhishekalbert committed:
Merge pull request #5 from oppia/develop

syncing the forked repo with original repo

abhishekalbert committed:
Fixed End-to-End Tests with action.js and waitFor.js

Fixed .click() call on an element to action.click in workflow.js ( core/tests/protractor_utils/workflow.js
) file

abhishekalbert committed:
Fixed lint issues

abhishekalbert committed:
fixed lint issues

abhishekalbert committed:
Added waitFor statements

abhishekalbert committed:
Improve the readbility and added action.sendKeys() as commented

abhishekalbert committed:
Edited assertion

abhishekalbert committed:
Final commit made all the changes accroding to comment.

abhishekalbert committed:
Merge pull request #6 from oppia/develop

syncing with main repo

abhishekalbert committed:
fix lint errors

abhishekalbert committed:
Merge branch 'e2eTestwithActionAndWaitfor' into develop

abhishekalbert committed:
Update question-backend-api.service.ts

abhishekalbert committed:
Merge pull request #8 from abhishekalbert/develop

develop pull

abhishekalbert committed:
Update workflow.js

abhishekalbert committed:
Update question-backend-api.service.ts

U8NWXD commented:
Deassigning myself since I'm no longer a codeowner

abhishekalbert committed:
Merge pull request #10 from oppia/develop

pull
